File: sources/pastebin/rxor.vb
    '------------------
    'Creator: aeonhack
    'Site: elitevs.net
    'Created: 9/20/2012
    'Changed: 9/20/2012
    'Version: 1.0.0
    '------------------
    Public Function RXOR(data As Byte(), key As Byte()) As Byte()
        Dim N1 As Integer = 11
        Dim N2 As Integer = 13
        Dim NS As Integer = 257

        For I As Integer = 0 To key.Length - 1
            NS += NS Mod (key(I) + 1)
        Next

        Dim T(data.Length - 1) As Byte
        For I As Integer = 0 To data.Length - 1
            NS = key(I Mod key.Length) + NS
            N1 = (NS + 5) * (N1 And 255) + (N1 >> 8)
            N2 = (NS + 7) * (N2 And 255) + (N2 >> 8)
            NS = ((N1 << 8) + N2) And 255

            T(I) = data(I) Xor CByte(NS)
        Next

        Return T
    End Function

File: sources/pastebin/rxor.cs
        //------------------
        //Creator: aeonhack
        //Site: elitevs.net
        //Created: 9/20/2012
        //Changed: 9/20/2012
        //Version: 1.0.0
        //------------------
        public byte[] RXOR(byte[] data, byte[] key)
        {
            int N1 = 11;
            int N2 = 13;
            int NS = 257;

            for (int I = 0; I <= key.Length - 1; I++)
            {
                NS += NS % (key[I] + 1);
            }

            byte[] T = new byte[data.Length];
            for (int I = 0; I <= data.Length - 1; I++)
            {
                NS = key[I % key.Length] + NS;
                N1 = (NS + 5) * (N1 & 255) + (N1 >> 8);
                N2 = (NS + 7) * (N2 & 255) + (N2 >> 8);
                NS = ((N1 << 8) + N2) & 255;

                T[I] = (byte)(data[I] ^ (byte)(NS));
            }

            return T;
        }